mysql-test/suite/wsrep/t/wsrep_set_var_extension.test

```diff
@@ -0,0 +1,130 @@
+# === Purpose ===
+#
+# This test verifies that the following wsrep session variables are hint updatable.
+#
+# 1. wsrep_sync_wait
+# 2. wsrep_dirty_reads
+# 3. wsrep_trx_fragment_size
+# 4. wsrep_trx_fragment_unit
+#
+# === References ===
+#
+# PXC-3743: Make wsrep session variables updatable through SET_VAR() hint
+
+--source include/have_wsrep_provider.inc
+
+# Auxiliary SP which returns variable value before hint, with hint and after hint applying
+DELIMITER \\; 
+
+CREATE PROCEDURE test_hint (hint_str VARCHAR(255), var_str VARCHAR(64))
+BEGIN
+
+SET @orig_q= CONCAT("SELECT VARIABLE_VALUE INTO @before_val FROM performance_schema.session_variables where VARIABLE_NAME = '",  var_str, "'");
+
+SET @hint_q= CONCAT("SELECT /*+ ", hint_str,
+                    "*/ VARIABLE_VALUE" ,
+                    " INTO @hint_val FROM performance_schema.session_variables where VARIABLE_NAME = '",  var_str, "'");
+
+SET @after_q= CONCAT("SELECT VARIABLE_VALUE INTO @after_val FROM performance_schema.session_variables where VARIABLE_NAME = '",  var_str, "'");
+
+PREPARE orig_stmt FROM @orig_q;
+PREPARE hint_stmt FROM @hint_q;
+PREPARE after_stmt FROM @after_q;
+
+EXECUTE orig_stmt;
+EXECUTE hint_stmt;
+EXECUTE after_stmt;
+
+SELECT  hint_str;
+SELECT @before_val, @hint_val, @after_val;
+
+DEALLOCATE PREPARE orig_stmt;
+DEALLOCATE PREPARE hint_stmt;
+DEALLOCATE PREPARE after_stmt;
+
+END\\
+
+DELIMITER ;\\
+
+--echo #########################################
+--echo # Test SET_VAR hint for wsrep variables #
+--echo #########################################
+
+--echo
+--echo # Test 1: wsrep_sync_wait
+CALL test_hint("SET_VAR(wsrep_sync_wait=4)", "wsrep_sync_wait");
```

Comment:
What is the purpose of this call?

Reply:
The intention here is to first test with the MTR's default value for the variables, then changing the value and testing again.

Reply:
OK, but shouldn't we do asserts as below? The stored procedure does not do any validation.

Reply:
We want to stop executing the test immediately our expectations are not met and this is not possible with the stored routines. So, the stored procedure only sets the uservars and they will be later used for the validation.

Reply:
I think I got your point now. You meant to add asserts even after the first call right?

Reply:
Fixed now.